apps/examples/aifw_test/aifw_test_main.cpp

```diff
@@ -0,0 +1,156 @@
+		AIFW_LOGE("AI helper stop failed");
+	}
+	ai_helper_deinit();
+	AIFW_LOGV("AI helper deinit done.");
+}
+
+/**
+ * @brief: AI Framework calls this function to collect the raw data and pass it for inference.
+ * This callback is called when timer expires. Time interval is set in 'inferenceInterval' field of json.
+*/
+static void sine_collectRawDataListener(void)
+{
+	memset(gSensorValues, '\0', gSensorValueCount * sizeof(float));
+	AIFW_RESULT result = readCSVData(gHandle, gSensorValues);
+	if (result == AIFW_OK) {
+		result = ai_helper_push_data(gSineWaveCode, (void *)gSensorValues, gSensorValueCount);
+		if (result == AIFW_OK) {
+			AIFW_LOGV("push data operation OK");
+		} else {
+			AIFW_LOGE("push data operation failed. ret: %d", result);
+		}
+		return;
+	}
+	AIFW_LOGE("reading input CSV data return error result : %d", result);
+	aifw_test_deinit();
+}
+
+/**
+ * @brief: Application will receive inference result in this function.
+ * It is mandatory to be defined by each application.
+ * @param [in] res: Successful inference operation 'res' is set to AIFW_OK. Errors are set as per AIFW_RESULT enum values.
+ * @param [in] values: inference result values after ensembling 
+ * @param [in] count: count of values in inference result
+*/
+static void sine_inferenceResultListener(AIFW_RESULT res, void *values, uint16_t count)
+{
+	if (res != AIFW_OK) {
+		AIFW_LOGE("Inference failed for this cycle, ret: %d", res);
+		return;
+	}
+	float *predictedResult = (float *)values;
+	/* Read expected inference result from result CSV to compare it from predicted inference result */
+	memset(gResultValues, '\0', gResultValueCount * sizeof(float));
+	AIFW_RESULT result = readCSVData(gResultHandle, gResultValues);
+	if (result != AIFW_OK) {
+		AIFW_LOGE("reading result CSV data failed, inference result values cannot be verfied for this inference cycle. ret: %d", result);
+	}
+	AIFW_LOGI("Expected value: %f, AIFW prediction result : %f", gResultValues[1], predictedResult[0]);
+}
+
+int aifw_test_main(int argc, char *argv[])
+{
+	/* Initialize CSV data source for input raw data */
+	AIFW_RESULT res = csvInit(&gHandle, "/mnt/AI/SineWave_packet.csv", FLOAT32, false);
+	if (res != AIFW_OK) {
+		AIFW_LOGE("FILE NOT FOUND || ERROR OPENING CSV. ret: %d", res);
+		return -1;
+	}
+	res = getColumnCount(gHandle, &gSensorValueCount);
+	if (res != AIFW_OK) {
+		AIFW_LOGE("Fetching sensor value count failed. ret: %d\n", res);
+		return -1;
+	}
+	gSensorValues = (float *)malloc(gSensorValueCount * sizeof(float));
+	if (!gSensorValues) {
+		AIFW_LOGE("Memory allocation failed for sensor values buffer");
+		return -1;
+	}
+	AIFW_LOGV("Raw input data csv initialization OK");
+
+	/* Initialize CSV data source for expected inference result data */
+	res = csvInit(&gResultHandle, "/mnt/AI/SineWave_resultPacket.csv", FLOAT32, false);
+	if (res != AIFW_OK) {
+		AIFW_LOGE("FILE NOT FOUND || ERROR OPENING CSV. ret: %d", res);
+		return -1;
```

Comment:
In this case, `gSensorValues` is allocated. But this line exits the function without freeing. Is it ok? No leak this case?
Same as other return lines.

Reply:
Memory leakage is now handled.